activerecord/test/cases/dup_test.rb

```diff
@@ -157,5 +158,34 @@ def test_dup_without_primary_key
         record.dup
       end
     end
+
+    def test_dup_transaction_state_and_transaction_record_state_are_cleared
+      topic = Topic.first
+      topic.send(:set_transaction_state, ActiveRecord::ConnectionAdapters::TransactionState.new(:rollback))
+      topic.send(:remember_transaction_record_state)
```

Comment:
We don't need to test internal states with private methods as long as it can be tested from public API.
And also, this test case does not show that the state of the record when the transaction was rolled back has been fixed.

Reply:
Thanks for review.
This test only checks that the transaction state is cleared in the duped record. I think that we must clear this state as well as other statuses. Do you think it is necessary to test the complete scheme with the rollback of the transaction?
And, unfortunately I did not find a public methods to access to `@transaction_state` and `@_start_transaction_state`, could you help me ?

Reply:
Since these are internal states we don't have any way to access to the states.
So it is enough to test the transaction behavior which tested in the following test rather than testing the states.

The `test_dup_record_not_persisted_after_rollback_transaction` sufficiently shows which behavior is fixed.
So we simply remove the `test_dup_transaction_state_and_transaction_record_state_are_cleared`.

Reply:
ok, I removed the test and added the entry to CHANGELOG.